internal/util/cache/lru_cache.go

```diff
@@ -164,49 +192,51 @@ func (c *LRU) Keys() []Key {
 	keys := make([]Key, len(c.items))
 	i := 0
 	for ent := c.evictList.Back(); ent != nil; ent = ent.Prev() {
-		keys[i] = ent.Value.(*entry).key
+		keys[i] = ent.Value.(*Entry).Key
 		i++
 	}
 	return keys
 }
 
-func (c *LRU) Len() int {
+func (c *LRU) Size() uint64 {
 	c.m.RLock()
 	defer c.m.RUnlock()
-	return c.evictList.Len()
+	return c.size
 }
 
-func (c *LRU) Capacity() int {
+func (c *LRU) Capacity() uint64 {
 	return c.capacity
 }
 
 func (c *LRU) Purge() {
 	c.m.Lock()
 	defer c.m.Unlock()
 	for k, v := range c.items {
-		if c.onEvicted != nil {
-			c.evictedCh <- v.Value.(*entry)
+		if c.storeHelper.OnEvicted != nil {
+			c.evictedCh <- v.Value.(*Entry)
 		}
 		delete(c.items, k)
 	}
 	c.evictList.Init()
+	c.size = 0
 }
 
-func (c *LRU) Resize(capacity int) int {
+func (c *LRU) Resize(capacity uint64) uint64 {
 	c.m.Lock()
 	defer c.m.Unlock()
 	c.capacity = capacity
-	if capacity >= c.evictList.Len() {
+	if capacity >= c.size {
 		return 0
 	}
-	diff := c.evictList.Len() - c.capacity
-	for i := 0; i < diff; i++ {
+	diff := c.size - capacity
+	for c.size > capacity {
 		oldestE := c.evictList.Back()
 		if oldestE != nil {
 			c.evictList.Remove(oldestE)
-			kv := oldestE.Value.(*entry)
-			delete(c.items, kv.key)
-			if c.onEvicted != nil {
+			kv := oldestE.Value.(*Entry)
+			delete(c.items, kv.Key)
+			c.size -= uint64(kv.Size)
```

Comment:
same here

Reply:
For measureSize may produce overhead, I hope it is used for once. So size has been assign into Entry in line 135.